frontend/packages/ux-editor/src/components/config/editModal/EditImage/ExternalImage/ExternalImageValidationStatus.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import classes from './ExternalImageValidationStatus.module.css';
+import { StudioParagraph, StudioSpinner } from '@studio/components';
+import { useTranslation } from 'react-i18next';
+import type { ExternalImageUrlValidationResponse } from 'app-shared/types/api/ExternalImageUrlValidationResponse';
+
+interface ExternalImageValidationStatusProps {
+  validationStatus: string;
+  validationResult: ExternalImageUrlValidationResponse;
+}
+
+export const ExternalImageValidationStatus = ({
+  validationStatus,
+  validationResult,
+}: ExternalImageValidationStatusProps) => {
+  const { t } = useTranslation();
+  const { getValidationMessageBasedOnValidationStatus } =
+    useValidationMessageBasedOnValidationStatus();
+
+  const validationMessage = getValidationMessageBasedOnValidationStatus(validationResult);
+
+  switch (validationStatus) {
+    case 'pending':
+      return (
+        <StudioSpinner
+          size='small'
+          showSpinnerTitle
+          spinnerTitle={t('ux_editor.properties_panel.images.validating_image_url_pending')}
+        />
+      );
+    case 'error':
+      return (
+        <StudioParagraph className={classes.validationStatusContainer} size='small'>
+          {t('ux_editor.properties_panel.images.validating_image_url_error')}
+        </StudioParagraph>
+      );
+    case 'success':
+      return validationMessage === '' ? null : (
+        <StudioParagraph className={classes.validationStatusContainer} size='small'>
+          {validationMessage}
+        </StudioParagraph>
+      );
+  }
+};
+
+export const useValidationMessageBasedOnValidationStatus = () => {
+  const { t } = useTranslation();
+
+  const getValidationMessageBasedOnValidationStatus = (
+    validationResultStatus: ExternalImageUrlValidationResponse,
+  ): string => {
+    switch (validationResultStatus) {
+      case 'Ok':
+        return '';
+      case 'NotValidUrl':
+        return t('ux_editor.properties_panel.images.invalid_external_url');
+      case 'NotAnImage':
+        return t('ux_editor.properties_panel.images.invalid_external_url_not_an_image');
+    }
+  };
+
+  return { getValidationMessageBasedOnValidationStatus };
+};
```

Comment:
Fin hook! Jeg tenker vi kan forenkle den litt. Den trenger ikke returnere et objekt, og `validationResultStatus` kommer fra propsene til komponenten, så den kan vi vel sende rett inn uten å måtte returnere en funksjon som blir kalt etterpå?

```suggestion
export const useValidationMessageBasedOnValidationStatus = (validationResultStatus: ExternalImageUrlValidationResponse): string => {
  const { t } = useTranslation();

  switch (validationResultStatus) {
    case 'Ok':
      return '';
    case 'NotValidUrl':
      return t('ux_editor.properties_panel.images.invalid_external_url');
    case 'NotAnImage':
      return t('ux_editor.properties_panel.images.invalid_external_url_not_an_image');
  }
};
```